File: 0682-baseball-game/0682-baseball-game.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> st = new Stack<>();
        for (String operation : operations) {
            if (Character.isDigit(operation.charAt(operation.length() - 1))) st.push(Integer.parseInt(operation));
            else if (operation.startsWith("C")) {
                st.pop();
            } else if (operation.startsWith("D")) {
                int x = st.peek();
                st.push(x * 2);
            } else {
                int x = st.pop();
                if (!st.isEmpty()) {
                    int y = st.pop();
                    st.push(y);
                    st.push(x);
                    st.push(x + y);
                } else {
                    st.push(x);
                    st.push(x);
                }
            }
            System.out.println(st);
        }
        int res = 0;
        while (!st.isEmpty()) {
            res += st.pop();
        }
        return res;
    }
}

File: 0682-baseball-game/0682-baseball-game.kt
class Solution {
    fun calPoints(operations: Array<String>): Int {
        val stack = Stack<Int>()
        operations.forEach { operation ->
            if (operation[operation.length - 1].isDigit()) stack.push(operation.toInt())
            else if (operation[0] == 'C') stack.pop()
            else if (operation[0] == 'D') {
                stack.push(stack.peek() * 2)
            } else {
                val x = stack.pop()
                if (!stack.isEmpty()) {
                    val y = stack.pop()
                    stack.push(y)
                    stack.push(x)
                    stack.push(x + y)
                } else 
                    stack.push(x)
            }
        }
        val res = stack.fold(0) { acc, i -> 
            acc + i
        }
        return res
    }
}